Improve locale handling. Don't try to do weird thing in the environment, but
select explicitly at the python level how (in which locale) we want to write on
stdout, or in the various opened files

--- doc/tools/generate_doc.orig	2013-04-02 14:40:29.000000000 +0200
+++ doc/tools/generate_doc	2013-05-29 09:59:30.674749162 +0200
@@ -55,7 +55,7 @@
                           stereo_unit, \
                           video_camera
 
-import sys, os
+import sys, os, codecs
 import fnmatch
 
 from morse.core.actuator import Actuator
@@ -63,6 +63,8 @@
 
 from morse.builder.data import MORSE_DATASTREAM_DICT
 
+sys.stdout = codecs.getwriter("utf-8")(sys.stdout.detach())
+
 PREFIX = "."
 MEDIA_PATH = "../../media"
 
@@ -487,7 +489,8 @@
 
 for name, props in components.items():
     module = (props['object'].__module__.split('.'))[-1]
-    with open(os.path.join(PREFIX, props['type'] + 's', module + ".rst"), 'w') as out:
+    with open(os.path.join(PREFIX, props['type'] + 's', module + ".rst"), 'w',
+              encoding='utf-8') as out:
         out.write(underline(name) + '\n')
 
         # if an image if available, use it
@@ -572,7 +575,7 @@
 def generate_matrix(filename):
     datastreams = ['text', 'socket', 'yarp', 'ros', 'pocolibs', 'moos']
 
-    with open(filename, 'w') as out:
+    with open(filename, 'w', encoding='utf-8') as out:
         sensors_list = []
         actuators_list = []
         for name, props in components.items():
